tests/Samples.Server.Tests/ResponseTests.cs

```diff
@@ -1,17 +1,56 @@
-﻿using GraphQL.Server.Transports.AspNetCore.Common;
-using Shouldly;
 using System.Threading.Tasks;
 using Xunit;
+using GraphQL.Server.Common;
+using GraphQL;
+
+#if NETCOREAPP2_2
+using GraphQL.NewtonsoftJson;
+#else
+using GraphQL.SystemTextJson;
+#endif
 
 namespace Samples.Server.Tests
 {
     public class ResponseTests : BaseTest
     {
-        [Fact]
-        public async Task Single_Query_Should_Return_Single_Result()
+        [Theory]
+        [InlineData(RequestType.Get)]
+        [InlineData(RequestType.PostWithJson)]
+        [InlineData(RequestType.PostWithGraph)]
+        [InlineData(RequestType.PostWithForm)]
+        public async Task Single_Query_Should_Return_Single_Result(RequestType requestType)
         {
-            var response = await SendRequestAsync(new GraphQLRequest { Query = "{ __schema { queryType { name } } }" });
-            response.ShouldBe(@"{""data"":{""__schema"":{""queryType"":{""name"":""ChatQuery""}}}}", ignoreExtensions: true);
+            var request = new GraphQLRequest { Query = "{ __schema { queryType { name } } }" };
+            var response = await SendRequestAsync(request, requestType);
+            response.ShouldBeEquivalentJson(@"{""data"":{""__schema"":{""queryType"":{""name"":""ChatQuery""}}}}", ignoreExtensions: true);
+        }
+
+        [Theory]
```

Comment:
Why is there no `[InlineData(RequestType.Get)]` ?

Reply:
Because this test explicitly is checking for POST scenarios that any query string values are pulled off and used instead of the body, if any. It doesn't make sense for a GET scenario. Have also updated the Send request method to throw if such a combination was done. Hopefully that's clearer.

Reply:
Now I got it. And I understand how important it is to write this explanation in the comments on this test.